### ISSUE TITLE
✍️ Contribution period: Deborah Oyawale 

### ISSUE THREAD

**Issue by Deborah-M-Oyawale** [ersilia-os/ersilia#1023]
### Week 1 - Get to know the community

- [x] Join the communication channels
- [x] Open a GitHub issue (this one!)
- [x] Install the Ersilia Model Hub and test the simplest model
- [x] Install Docker if needed, and test another model
- [x] Write a motivation statement to work at Ersilia
- [x] Submit your first contribution to the Outreachy site

### Week 2 - Get Familiar with Machine Learning for Chemistry

- [x] Select a model from the list suggested in GitBook
- [x] Download and serve the model via the Ersilia Model Hub to ensure it works
- [ ] Open a repository on your GitHub user with all the necessary files
- [x] Select and clean a dataset of 1000 molecules (example notebook 1)
- [ ] Run predictions for the molecules on the selected model and evaluate the results

### Week 3 - Validate a Model in the Wild

- [ ] Find a suitable dataset with sufficient experimental results
- [ ] Clean and standardize the dataset
- [ ] Run predictions and calculate metrics.

### Week 4 - Prepare your final application

- [ ] Submit the final application in the Outreachy website

**Comment by Deborah-M-Oyawale** [comment]
WEEK 1 (TASK1)

1. I read through the information provided about Ersilia and was fascinated about the AI/ML models being worked on for biomedical research and predictions.
2. I joined the slack channel and introduced myself on the #general channel as instructed.
3. Went through the repo on GitHub while familiarizing myself with the issues being worked on the progress made thus far on Ersilia Model Hub.
4. Opened the Outreachy Summer 24 issue.
5. Using a windows laptop, I had to set up my system by installing WSL. Following the instruction and guide in the Ersilia Book, by running `wsl --install` on my terminal. This automatically installed the ubuntu terminal on my machine. Having done this, I moved on to installing the pre-requisites on my laptop, to enable the successful installation of the Ersilia tool.
6. Pre-requisite 1 `sudo apt install build-essential` and Pre-requisite 2 worked fine. I encountered a minor issue with Pre-requisite 3 `gh auth login`. I couldn't login to git directly from the  terminal. I resorted to pasting the link: https://github.com/login/device into my browser and authenticating my login from there. Pre-requisite 4 installed and successfully activated Git LFS.
7. Pre-requisite 5 which is installation of Isaura was outputting errors: 'Failed building wheel for h5py' 
'Could not build wheels for h5py, which is required to install pyproject.toml-based projects'. While debugging, I decided to install and activate Ersilia first. I encountered this error: 'ersilia does not appear to be a python project: neither 'setup.py' nor  'pyproject.toml' found' when I tried installing pip with the command line: `pip install -e`. While debugging and trying everything possible that was suggested on the #debugging channel by my peers, all to no avail. I decided to switch the terminal from linus to Ubuntu terminal, and I then started again from the first instruction Ersilia Book. the installation then successful afterwards. I proceeded to pre-requisite 5 again, which was successful afterwards.
8. I proceeded to pre-requisite 6: Installation of Docker on Ubuntu.
9. I then tested a simple model```
 ersilia -v fetch eos3b5e
ersilia serve eos3b5e
ersilia -v api run -i "CCCC"
``` which calculated the molecular weight of the molecules and printed the result in my CLI and I also tested ersilia with docker which gave the expected output.

**Comment by Deborah-M-Oyawale** [comment]
**MOTIVATION STATEMENT LETTER**

My name is Deborah Oyawale, and I am writing to express my motivation for joining Outreachy and my genuine enthusiasm and eagerness to contribute to Ersilia Open Source Initiative.

As a young lady who has been trying to transition into the tech space, having gained vast knowledge in data science and worked on several datasets from Kaggle.com. I have been able to analyze several datasets and build different ML models. One of the projects which stood out for me was the titanic ship, where I was able to build models and make predictions with a minimum of 70% accuracy. Ever since then, I have been eager to work with ML/AI to make valuable predictions that can at least save lives. 

It has been difficult to get a seat at the table to at least prove my competence. Having come across Outreachy, I was super duper excited of the possibility of being seen and given a fair chance to contribute to the development of AI/ML models as an aspiring data scientist. The thought of me being able to work on a real life project and advance my career in the tech space is  my motivation for joining Outreachy.

Over the course of my academic and professional journey, I have developed a deep passion for artificial intelligence and machine learning. The thought of harnessing the power of data to create intelligent systems that can make a positive impact on society and save lives has been a driving force in my career aspirations. Getting into the Outreachy contribution phase was really elating for me as it seemed as though I’m a step closer to achieving my dreams. 

Ersilia project has captured my attention for a couple of reasons. Firstly, the innovative approach of developing AI/ML models taken towards solving biomedical problems aligns with my own vision for the potential of AI/ML in proffering solutions in the health sector and saving lives. Also, the collaborative and open nature of Ersilia community (which I have experienced firsthand on the #debugging channel) fosters an environment of shared knowledge and continuous learning, which I find incredibly valuable.
I believe that contributing to Ersilia Model Hub will not only allow me to expand my already honed technical skills in data science with vast knowledge working with Python, GitHub and a conceptual knowledge of Docker but also provide me with an opportunity to work alongside some of the brightest minds in the tech industry on a quest of developing and implementing artificial intelligence and machine learning tools, that are focused in drug discovery against infectious and neglected diseases, while indirectly saving lives and revolutionizing biomedical practice. I am confident that my unique perspective and dedication to excellence will contribute positively to the Ersilia’s objectives.
During the internship, my primary goal is to actively contribute to Ersilia's ongoing projects while learning about the field of ML applied to drug discovery and the basics of ML deployment through Docker from Ersilia’s mentors: Gemma Turon, Miquel Duran-Frigola, Dhanshree Arora and Ini-Abasi Bernard. I plan to engage with the community, seek mentorship, and collaborate with fellow contributors in expanding the collection of AI/ML models available and extending the capacities of Ersilia Model Hub platform to enhance its performance, while maximizing my learning experience. Post-internship, I aim to continue contributing to Ersilia and leverage the skills gained to further community-centric projects in my future career in the health sector as a data scientist.
In conclusion, I am genuinely enthusiastic about the opportunity to be part of Outreachy and work with the Ersilia team. I am ready to commit my time, energy, and skills to support the growth and success of this project.

Thank you for considering my application. I am excited about the potential to collaborate and contribute to the future of AI/ML in biomedical practice through Ersilia Model Hub.

**Comment by GemmaTuron** [comment]
Hi @Deborah-M-Oyawale 

Please let us know if you still plan to make an application to Ersilia, otherwise we will close this issue so mentors can focus on supporting contributors who want to make a final application.

Thansk for your understanding

**Comment by Deborah-M-Oyawale** [comment]
> Hi @Deborah-M-Oyawale
> 
> Please let us know if you still plan to make an application to Ersilia, otherwise we will close this issue so mentors can focus on supporting contributors who want to make a final application.
> 
> Thansk for your understanding

Hi @GemmaTuron thank you for reaching out. I’ve been in and out of the hospital for over two weeks now because of my health. This has slowed down my pace as I am not 100% fit yet. I recorded my first application to Ersilia on Outreachy website. I intend getting back on track this weekend after completing my medication.

I’m sorry if I have caused any delay to the company organizers.

**Comment by GemmaTuron** [comment]
Hi @Deborah-M-Oyawale 

Sorry to hear you are not feeling well. Please let me know once you are ready to continue your contribution

**Comment by Deborah-M-Oyawale** [comment]
> Hi @Deborah-M-Oyawale
> 
> Sorry to hear you are not feeling well. Please let me know once you are ready to continue your contribution

Good morning Gemma. Trust you are well?
I'm back to working on my project, but I know I can't meet up again

Can I still continue working on the project? I really just want to try my hands on it till the very end. Is that fine?

because I have never worked on a biomedical project before
